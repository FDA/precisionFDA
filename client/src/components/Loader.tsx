/* eslint-disable react/require-default-props */
import React from 'react'
import styled, { css } from 'styled-components'
import { colors } from '../styles/theme'
import { Svg } from './icons/Svg'

const LoaderWrapper = styled.div<{shouldDisplayInline?: boolean}>`
  display: flex;
  justify-content: center;
  margin-top: 16px;
  ${({ shouldDisplayInline }) => shouldDisplayInline && css`
    display: inline;
  `}
`
type Props = {
  height?: number
  shouldDisplayInline?: boolean
}

export const Loader = ({ height = 16, shouldDisplayInline }: Props) => (
  <LoaderWrapper shouldDisplayInline={shouldDisplayInline}>
    <Svg height={height} width={60}>
      <circle fill={colors.darkBlue} stroke="none" cx="6" cy={height / 2} r="6">
        <animate
          attributeName="opacity"
          dur="1s"
          values="0;1;0"
          repeatCount="indefinite"
          begin="0.1"
        />
      </circle>
      <circle fill={colors.darkBlue} stroke="none" cx="26" cy={height / 2} r="6">
        <animate
          attributeName="opacity"
          dur="1s"
          values="0;1;0"
          repeatCount="indefinite"
          begin="0.2"
        />
      </circle>
      <circle fill={colors.darkBlue} stroke="none" cx="46" cy={height / 2} r="6">
        <animate
          attributeName="opacity"
          dur="1s"
          values="0;1;0"
          repeatCount="indefinite"
          begin="0.3"
        />
      </circle>
    </Svg>
  </LoaderWrapper>
)
