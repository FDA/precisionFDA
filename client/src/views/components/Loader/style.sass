.pfda-loader
  font-size: 36px
