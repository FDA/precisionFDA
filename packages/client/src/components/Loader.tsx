/* eslint-disable react/require-default-props */
import React from 'react'
import styled from 'styled-components'
import { Svg } from './icons/Svg'

export const LWrap = styled.div`
  align-self: stretch;
  height: 100%;
`

export const LoaderMargin = styled.div`
  margin: 24px;
`

const LoaderWrapper = styled.div`
  display: flex;
  justify-content: center;
  &.inline {
    display: inline;
  }
  &.pageloader {
    padding-top: 16px;
  }
`
type Props = {
  height?: number
  className?: string
}

export const Loader = ({ height = 16, ...rest }: Props) => {
  // Radius of circles; Max 6px, but smaller in small loaders
  const radius = height >= 12 ? 6 : height / 2

  // Width; max 60px
  const width = radius * 10

  // Horizontal position of circles
  const cx1 = 6 * width / 60
  const cx2 = 26 * width / 60
  const cx3 = 46 * width / 60

  return (
    <LoaderWrapper {...rest}>
      <Svg height={height} width={width}>
        <circle fill="var(--base)" stroke="none" cx={cx1} cy={height / 2} r={radius}>
          <animate
            attributeName="opacity"
            dur="1s"
            values="0;1;0"
            repeatCount="indefinite"
            begin="0.1"
          />
        </circle>
        <circle fill="var(--base)" stroke="none" cx={cx2} cy={height / 2} r={radius}>
          <animate
            attributeName="opacity"
            dur="1s"
            values="0;1;0"
            repeatCount="indefinite"
            begin="0.2"
          />
        </circle>
        <circle fill="var(--base)" stroke="none" cx={cx3} cy={height / 2} r={radius}>
          <animate
            attributeName="opacity"
            dur="1s"
            values="0;1;0"
            repeatCount="indefinite"
            begin="0.3"
          />
        </circle>
      </Svg>
    </LoaderWrapper>
  )
}
