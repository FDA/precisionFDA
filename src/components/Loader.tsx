/* eslint-disable react/require-default-props */
import React from 'react'
import styled, { css } from 'styled-components'
import { colors } from '../styles/theme'
import { Svg } from './icons/Svg'

const LoaderWrapper = styled.div<{displayInline?: boolean}>`
  display: flex;
  justify-content: center;
  padding-top: 16px;
  ${({ displayInline }) => displayInline && css`
    display: inline;
  `}
`
type Props = {
  height?: number
  displayInline?: boolean
}

export const Loader = ({ height = 16, displayInline }: Props) => (
  <LoaderWrapper displayInline={displayInline}>
    <Svg height={height} width={60}>
      <circle fill={colors.darkBlue} stroke="none" cx="6" cy={height / 2} r="6">
        <animate
          attributeName="opacity"
          dur="1s"
          values="0;1;0"
          repeatCount="indefinite"
          begin="0.1"
        />
      </circle>
      <circle fill={colors.darkBlue} stroke="none" cx="26" cy={height / 2} r="6">
        <animate
          attributeName="opacity"
          dur="1s"
          values="0;1;0"
          repeatCount="indefinite"
          begin="0.2"
        />
      </circle>
      <circle fill={colors.darkBlue} stroke="none" cx="46" cy={height / 2} r="6">
        <animate
          attributeName="opacity"
          dur="1s"
          values="0;1;0"
          repeatCount="indefinite"
          begin="0.3"
        />
      </circle>
    </Svg>
  </LoaderWrapper>
)
