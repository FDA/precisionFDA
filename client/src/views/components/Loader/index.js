import React from 'react'

import Icon from '../Icon'
import './style.sass'


const Loader = () => (
  <Icon icon="fa fa-spinner fa-spin" cssClasses="pfda-loader" />
)

export default Loader
