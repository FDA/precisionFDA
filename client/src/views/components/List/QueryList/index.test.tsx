import React from 'react'
import { shallow, mount } from 'enzyme'

import Loader from '../../Loader'
import { QueryList } from '.'

const getMockData = () => {
  const mockData: any[] = []
  for (let i=0; i<10; i++) {
    mockData.push({
      id: i,
      name: 'Name'+i,
      description: 'Blah '+i
    })
  }
  return mockData
}

const getMockQuerySuccess = () => {
  const status = 'success'
  const error = undefined
  const data = getMockData()
  return { status, error, data }
}

const listExtractor = (data: any) => {
  return data
}

const template = (item: any) => {
  return (
    <div key={item.id}>
      <h1>{item.name}</h1>
      <p>{item.description}</p>
    </div>
  )
}

describe('QueryList test', () => {
  it('should render', () => {
    const wrapper = shallow(<QueryList query={getMockQuerySuccess} listExtractor={listExtractor} template={template} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should show loader when isFetching and not show QueryList', () => {
    const query = () => {
      const status = 'loading'
      const error = undefined
      const data = getMockData()
      return { status, error, data }
    }
    const wrapper = mount(<QueryList query={query} listExtractor={listExtractor} template={template} />)

    expect(wrapper.find(Loader)).toHaveLength(1)
  })
})
