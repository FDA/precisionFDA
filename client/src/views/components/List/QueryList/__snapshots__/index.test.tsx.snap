// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryList test should render 1`] = `
<ul
  className="query-list"
>
  <div
    key="0"
  >
    <h1>
      Name0
    </h1>
    <p>
      Blah 0
    </p>
  </div>
  <div
    key="1"
  >
    <h1>
      Name1
    </h1>
    <p>
      Blah 1
    </p>
  </div>
  <div
    key="2"
  >
    <h1>
      Name2
    </h1>
    <p>
      Blah 2
    </p>
  </div>
  <div
    key="3"
  >
    <h1>
      Name3
    </h1>
    <p>
      Blah 3
    </p>
  </div>
  <div
    key="4"
  >
    <h1>
      Name4
    </h1>
    <p>
      Blah 4
    </p>
  </div>
  <div
    key="5"
  >
    <h1>
      Name5
    </h1>
    <p>
      Blah 5
    </p>
  </div>
  <div
    key="6"
  >
    <h1>
      Name6
    </h1>
    <p>
      Blah 6
    </p>
  </div>
  <div
    key="7"
  >
    <h1>
      Name7
    </h1>
    <p>
      Blah 7
    </p>
  </div>
  <div
    key="8"
  >
    <h1>
      Name8
    </h1>
    <p>
      Blah 8
    </p>
  </div>
  <div
    key="9"
  >
    <h1>
      Name9
    </h1>
    <p>
      Blah 9
    </p>
  </div>
</ul>
`;
